src/main/java/commands/Command.java

```diff
@@ -13,48 +16,118 @@
     protected static final String CALC_REMAIN = "remain";
     protected static final String LIST = "list";
     protected static final String EXIT = "bye";
+    protected String userInput;
     protected String replyMsg;
+    protected Parser parser = new Parser();
     protected UI ui = new UI();
+    protected PersonList personList = new PersonList();
 
-    public Command() {
+    public Command(String userInput) {
+        this.userInput = userInput;
     }
 
     /**
      * Executes the instruction according to the input after parsing
-     * @param Instr the instruction after parsing by parser
      * @throws UnknownException if the instruction cannot be executed
     */
-    /**
-    public void execute(Parser Instr) throws UnknownException {
-        //parser
-        try () {
-            switch () {
+
+    public void execute() throws UnknownException {
+        try {
+            switch (parser.parseKeyword(userInput)) {
             case ADD_PERSON_CMD:
-                //this.replyMsg =
+                addPerson();
                 break;
             case DELETE_PERSON_CMD:
+                deletePerson();
                 break;
             case ADD_INCOME_CMD:
+                addIncome();
                 break;
             case DELETE_INCOME_CMD:
+                deleteIncome();
                 break;
             case ADD_SPENT_CMD:
+                addSpend();
                 break;
             case DELETE_SPEND_CMD:
+                deleteSpend();
                 break;
             case CALC_REMAIN:
+                personList.getRemain();
                 break;
             case LIST:
+                personList.list();
                 break;
             case EXIT:
+                ui.exit();
+                System.exit(0);
                 break;
             default:
                 throw new UnknownException();
             }
         } catch (UnknownException e) {
             this.replyMsg = e.toString();
         }
-        GUI.printMsg(replyMsg);
+        ui.printMsg(replyMsg);
     }
-    */
+
+    private void deleteSpend() {
+        Person newPerson;
+        String uid;
+        int index;
+        uid = parser.parseUserIndex(userInput);
+        index = parser.parseRecIndex(userInput);
```

Comment:
I think it'll be cleaner if you can initialise these 3 variables in the same line which you declared them.
Similar for functions below